src/compiler/checker.ts

```diff
@@ -10702,8 +10779,24 @@ namespace ts {
                 }
 
                 let derived = getTargetSymbol(getPropertyOfObjectType(type, base.name));
-                if (derived) {
-                    let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+                let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+
+                Debug.assert(!!derived, "derived should point to something, even if it is the base class' declaration.");

```

Comment:
Separate issue? I think this PR might already be a good example of making too many non-essential changes, harming the readability of the change.